workspaces/util/src/chunkBySize.ts

```diff
@@ -1,7 +1,15 @@
 export function* chunkBySize<T>(
-  array: readonly T[],
+  array: readonly T[] | null | undefined,
   chunkSize: number,
 ): Generator<T[], void, void> {
+  if (chunkSize < 1 || !Number.isInteger(chunkSize)) {
+    throw new RangeError("Chunk size must be a positive integer!");
+  }
```

Comment:
I think it would also be ok to only check for >=1 here, and do `i += Math.trunc(chunkSize)` below. That would be consistent with how the built-in `.slice` works. But I'm ok with being stricter here if you prefer 😄